app/content/react/library-reference/api.md

```diff
@@ -62,22 +62,26 @@ React hook that allows a more powerful use of CARTO deck.gl layers, creating a s
 
 - **Input**:
 {{% tableWrapper tab="true" %}}
-| Param              | Type                | Description              |
-| ------------------ | ------------------- | ------------------------ |
-| props              | <code>Object</code> | Object properties        |
-| props.source       | <code>Object</code> | Data source              |
-| props.source.id    | <code>string</code> | Unique source ID         |
-| props.source.type  | <code>string</code> | Source type. Check available types [here](/deck-gl/reference#type-string)  |
-| props.source.connection  | <code>string</code> | Connection name. Required only for CARTO 3.  |
-| props.source.data  | <code>string</code> |  Table name, tileset name or SQL query                                           |
-| props.source.credentials | <code>string</code> |  Credentials for accessing the source (check the parameters [here](/deck-gl/reference#setdefaultcredentials))                                           |
-| [props.uniqueIdProperty] | <code>string</code> | (optional) Name of the column for identity. To be used internally when getting viewportFeatures (used by widget computations) |
-| [props.viewportFeatures] | <code>boolean</code> | (optional) Default: `true`. Whether to calculate viewport features for this layer or not.  |
-| [props.viewportFeaturesDebounceTimeout] | <code>number</code> | (optional) Default: `500`. Timeout for calculating viewport features when there's a change. It's used to avoid repeated viewport calculations in a short amount of time.  |
+| Param              | Type                | Default       | Description                                    |
+| ------------------ | ------------------- | ------------- | ---------------------------------------------- |
+| props              | <code>Object</code> |               | { source, [layerConfig], [uniqueIdProperty], [viewportFeatures], [viewportFeaturesDebounceTimeout]}  |
```

Comment:
Do we explicitly say anywhere that [aParam] in [] is optional? 

At the beginning it was a bit difficult to me to read properly expressions like `{ source, [layerConfig] }`  (I was basically thinking about destructuring in JS...)

Reply:
Not explicitly but it is the convention we are using always for the optional parameters. If you scroll below, you will see the layer parameters are defined as `(optional)` in the description.